stage2/Calculator.js

```diff
@@ -0,0 +1,170 @@
+document.addEventListener('DOMContentLoaded', function() {
+  var myCalc = new Calculator(document.querySelector('.calculator'))
+})
+
+// class definition for Calculator
+function Calculator(rootDOMNode){
+  this.operators = []
+  this.operands = []
+  this.latestCharClicked = undefined
+  this.rootDOMNode = rootDOMNode
+  this.onClick = this.onClick.bind(this)
+  this.bindEventListeners()
+  this.expressionPrecedence  = {
+    '-': 0,
+    '+': 0,
+    '/': 1,
+    'X': 1
+  }
+}
+
+// front-end browser stuff
+Calculator.prototype.bindEventListeners = function(){
+  const buttons = this.rootDOMNode.querySelectorAll('.calculator-button')
+  for (var i = 0; i < buttons.length; i++) {
+    buttons[i].addEventListener('click', this.onClick)
+  }
+}
+
+Calculator.prototype.onClick = function(event){
+  var clickedNodeData = event.target.innerText
+
+  if( (event.target.classList).contains('calculator-op') ){
+    this.pushOperand( clickedNodeData )
+  } else {
+    this.pushOperator( clickedNodeData )
+  }
+  // check special cases to force evaluation after a stack hits a certian size
+  this.rootDOMNode.querySelector('.calculator-screen').innerText = this.operands[this.operands.length - 1] || 0
+  this.checkStacks()
+}
+
+// Calculator work stuff
+Calculator.prototype.checkStacks = function() {
+
+}
+
+Calculator.prototype.pushOperand = function( value ) {
+  if( this.expressionPrecedence[value] < this.expressionPrecedence[this.operators[0]] ){
+    this.evaluateExpression()
+  }
+
+  switch( value ) {
+    case '=':
+      this.evaluateExpression()
+      break
+
+    case '+/-':
+      this.negateTopOperand()
+      break
+
+    case 'AC':
+      this.clearAll()
+      break
+
+    case 'C':
+      this.removeLatestOperand()
+      break
+
+    case '%':
+      this.percentigizeTopOperand()
+      break
+
+    default:
+      this.operators.push( value )
```

Comment:
I would validate that the value is a valid operator. There should be a fixed set of valid operators.